src/client/Microsoft.Identity.Client/AppConfig/ConfidentialClientApplicationBuilder.cs

```diff
@@ -152,6 +152,24 @@ public ConfidentialClientApplicationBuilder WithClientAssertion(string signedCli
             return this;
         }
 
+        /// <summary>
+        /// Sets the application client assertion. See https://aka.ms/msal-net-client-assertion
+        /// </summary>
+        /// <param name="clientAssertionDelegate">delegate computing the client assertion used to prove the identity of the application to Azure AD.
+        /// This is a delegate that computes a Base-64 encoded JWT for each authentication call.</param>
+        /// <returns>The ConfidentialClientApplicationBuilder to chain more .With methods</returns>
+        public ConfidentialClientApplicationBuilder WithClientAssertion(Func<string> clientAssertionDelegate)
```

Comment:
I would add a comment on the old `WithClientAssertion(string)` as well to tell users that if they want to cache their assertions, they can use this method.